src/Nethermind/Nethermind.Evm/Tracing/GethStyle/GethLikeTxMemoryTracer.cs

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Core;
+using Nethermind.Int256;
+
+namespace Nethermind.Evm.Tracing.GethStyle;
+
+public class GethLikeTxMemoryTracer : GethLikeTxTracer<GethTxMemoryTraceEntry>
+{
+    public GethLikeTxMemoryTracer(GethTraceOptions options) : base(options) => IsTracingMemory = IsTracingFullMemory;
+
+    public override void MarkAsSuccess(Address recipient, long gasSpent, byte[] output, LogEntry[] logs, Keccak? stateRoot = null)
+    {
+        base.MarkAsSuccess(recipient, gasSpent, output, logs, stateRoot);
+
+        Trace.Gas = gasSpent;
+    }
+
+    public override void SetOperationStorage(Address address, UInt256 storageIndex, ReadOnlySpan<byte> newValue, ReadOnlySpan<byte> currentValue)
+    {
+        base.SetOperationStorage(address, storageIndex, newValue, currentValue);
+
+        byte[] bigEndian = new byte[32];
```

Comment:
stackalloc?

Reply:
Could be. I haven't touched the existing code if not needed for this PR.